src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -222,13 +223,34 @@ case object TChar extends Type {
     .map(s => s.substring(0, 1))
 }
 
-abstract class TNumeric extends Type
+object TNumeric {
+  def parseResultType(types: Set[TNumeric]): Type = {
+    if (types(TDouble) || (types(TFloat) && (types(TInt) || types(TLong))))
+      TDouble
+    else if (types(TInt) && types(TLong))
+      TLong
```

Comment:
There probably is an easier way to do this, though -- promotion rules isn't a new problem